hassio/src/addon-view/hassio-addon-audio.ts

```diff
@@ -0,0 +1,228 @@
+            >
+              ${this.inputDevices &&
+                this.inputDevices.map((item) => {
+                  return html`
+                    <paper-item device="${item.device}"
+                      >${item.name}</paper-item
+                    >
+                  `;
+                })}
+            </paper-listbox>
+          </paper-dropdown-menu>
+          <paper-dropdown-menu
+            label="Output"
+            @value-changed="${this.setOutputDevice}"
+          >
+            <paper-listbox
+              slot="dropdown-content"
+              attr-for-selected="device"
+              selected="${this.selectedOutput}"
+            >
+              ${this.outputDevices &&
+                this.outputDevices.map((item) => {
+                  return html`
+                    <paper-item device="${item.device}"
+                      >${item.name}</paper-item
+                    >
+                  `;
+                })}
+            </paper-listbox>
+          </paper-dropdown-menu>
+        </div>
+        <div class="card-actions">
+          <mwc-button @click=${this._saveSettings}>Save</mwc-button>
+        </div>
+      </paper-card>
+    `;
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      haStyle,
+      hassioStyle,
+      css`
+        :host,
+        paper-card,
+        paper-dropdown-menu {
+          display: block;
+        }
+        .errors {
+          color: var(--google-red-500);
+          margin-bottom: 16px;
+        }
+        paper-item {
+          width: 450px;
+        }
+        .card-actions {
+          text-align: right;
+        }
+      `,
+    ];
+  }
+
+  protected update(changedProperties: PropertyValues): void {
+    super.update(changedProperties);
+    if (changedProperties.has("addon")) {
+      this.addonChanged();
+    }
+  }
+
+  protected setInputDevice(ev: PolymerChangedEvent<string>): void {
+    const value = ev.detail.value;
+    let device: undefined | string;
+    if (value) {
+      this.inputDevices!.map((item) => {
+        if (item.name === value) {
+          device = item.device;
+        }
+      });
+    }
+    if (device && device !== this.selectedInput) {
+      this.selectedInput = device;
+    }
+  }
+
+  protected setOutputDevice(ev: PolymerChangedEvent<string>): void {
+    const value = ev.detail.value;
+    let device: undefined | string;
+    if (value) {
+      this.outputDevices!.map((item) => {
+        if (item.name === value) {
+          device = item.device;
+        }
+      });
+    }
+    if (device && device !== this.selectedOutput) {
+      this.selectedOutput = device;
+    }
+  }
+
+  protected addonChanged(): void {
+    this.selectedInput = this.addon.audio_input || "null";
+    this.selectedOutput = this.addon.audio_output || "null";
+    if (this.outputDevices) {
+      return;
+    }
+
+    const noDevice = [{ device: "null", name: "-" }];
+    this.hass.callApi("GET", "hassio/hardware/audio").then(
```

Comment:
Convert this to async await

Reply:
Should I do it or not?
<https://github.com/home-assistant/home-assistant-polymer/pull/4398#discussion_r362412480>

Reply:
What you want :-)